apps/projects/app/components/Panel/NewIssue/AuthorizeGitHub.js

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useAragonApi } from '../../../api-react'
+import { Button, GU, Text } from '@aragon/ui'
+import { LoadingAnimation } from '../../Shared'
+import {
+  REQUESTED_GITHUB_TOKEN_SUCCESS,
+  REQUESTED_GITHUB_TOKEN_FAILURE,
+} from '../../../store/eventTypes'
+import { getToken, githubPopup, STATUS } from '../../../utils/github'
+
+const Loading = () => (
+  <div css={`text-align: center; margin-top: ${15 * GU}px`}>
+    <LoadingAnimation />
+  </div>
+)
+
+const SCOPE = 'public_repo'
+let popupRef = null
+
+export default function AuthorizeGitHub() {
+  const { api } = useAragonApi()
+  const [ githubLoading, setGithubLoading ] = useState(false)
+
+  useEffect(() => {
+    window.addEventListener('message', handlePopupMessage)
+    return () => {
+      window.removeEventListener('message', handlePopupMessage)
+    }
+  })
+
+  const handlePopupMessage = useCallback(async message => {
+    if (!popupRef) return
+    if (message.source !== popupRef) return
+
+    popupRef = null
+
+    if (message.data.from !== 'popup') return
+    if (message.data.name === 'code') {
+      const code = message.data.value
+      try {
+        const token = await getToken(code)
+        api.emitTrigger(REQUESTED_GITHUB_TOKEN_SUCCESS, {
+          status: STATUS.AUTHENTICATED,
+          scope: SCOPE,
+          token
+        })
+      } catch (err) {
+        api.emitTrigger(REQUESTED_GITHUB_TOKEN_FAILURE, {
+          status: STATUS.FAILED,
+          scope: null,
+          token: null,
+        })
+      }
+    }
+  }, [])
+
+  const requestExtraPermissions = useCallback(() => {
+    setGithubLoading(true)
+    popupRef = githubPopup(popupRef, SCOPE)
+  }, [])
+
+  if (githubLoading) return <Loading />
+
+  return (
+    <>
+      <br />
+      <Text.Block size="xlarge">We’re going to need more permissions</Text.Block>
+      <br />
+      <Text.Block>
+        The Projects app can make GitHub issues on your behalf, but to do that we need you to grant us more permissions.
+        <br />
+        <br />
+        We don’t think apps should ask for more permissions than they need. If you’re ready to start using the Projects app to make issues, we’ll request the minimal set of permissions that we can (GitHub call this the “<a href="https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes"><code>public_repo</code> scope</a>”).
```

Comment:
> If you’re ready to start using the Projects app to make issues, we’ll request the minimal set of permissions that we can

Something bothers me about this sentence. I think it's because of the phrase 'that we can' at the end. Maybe it would be better like "If you’re ready to start using the Projects app to make issues, we’ll request the minimal set of permissions that we need" or "If you’re ready to start using the Projects app to make issues, we’ll request the minimal set of permissions necessary to do so"? I'm not sure, maybe @stellarmagnet or @jayalaves  can weigh in.

Also:

> GitHub call this

GitHub calls this

Reply:
What I learned (this might be [wrong](https://www.quora.com/When-referring-to-a-company-name-as-the-subject-of-a-sentence-which-is-grammatically-correct-do-or-does)):

* "GitHub call this", treating companies/brands/bands as a collective, is the European version.
* "GitHub calls this", treating companies/brands/bands as an individual is the North American version.

Anyhow, it has been reworded!